### ISSUE TITLE
hGetBufSome: resource exhausted (Not enough space)

### ISSUE THREAD

**Issue by stianlagstad** [jgm/pandoc-citeproc#207]
I'm getting an error trying to use pandoc-citeproc on a x64 Windows 7 system:

`pandoc-citeproc --version`
> pandoc-citeproc 0.9

`pandoc-citeproc --help`
> Usage: pandoc-citeproc [options] [file..]
>   -h         --help           show usage information
>   -V         --version        show program version
>   -y         --bib2yaml       convert bibliography to YAML
>   -j         --bib2json       convert bibliography to JSON
>   -f FORMAT  --format=FORMAT  bibliography format

`pandoc-citeproc`
> pandoc-citeproc.exe: <stdin>: hGetBufSome: resource exhausted (Not enough space)

`pandoc --version`
> pandoc.exe 1.16.0.2

I installed both pandoc and pandoc-citeproc with `cabal install pandoc pandoc-citeproc`.
I found a related issue on pandoc here: <https://github.com/jgm/pandoc/issues/874>

Any idea what could be causing this?

-update:

I'm positive this didn't work earlier:
`pandoc essay.md -o essay.pdf --toc -N --filter pandoc-citeproc --csl style.csl`
but now it does.

`pandoc-citeproc`
Still gives the same error.

**Comment by jgm** [comment]
So this arises if you do `pandoc-citeproc` with no arguments?  Do you then hit `^Z` or does it immediately come up with the error?

jgm/pandoc#874 was solved by using strict instead of lazy IO.  So this might be a lazy IO issue.

If you look at Text.Pandoc.JSON in pandoc-types, you'll see that it uses lazy IO.  pandoc-citeproc uses this `toJSONFilter` function.

My guess is that replacing
```haskell
BL.getContents >>=
```
in these instances with
``` haskell
((fromChunks . (:[])) <$> B.getContents) >>=
```

might solve it.

**Comment by jgm** [comment]
Does anyone else see this using pandoc-citeproc on Windows?  It would be good to get some idea of how widespread/reproducible the problem is.  Please give Windows version + archictecture, as well as pandoc-citeproc version.

**Comment by stianlagstad** [comment]
It arises when I do `pandoc-citeproc` with no arguments, and it happens immediately without me doing anything.

I'm not sure how to test that change myself, but I'll update here if I figure out how.

**Comment by jgm** [comment]
+++ Stian Lågstad [Jan 17 16 04:00 ]:
>   It arises when I do pandoc-citeproc with no arguments, and it happens
>   immediately without me doing anything.
>
>   I'm not sure how to test that change myself, but I'll update here if I
>   figure out how.

If you want to try it, here's how.

Install stack for Windows.

https://github.com/commercialhaskell/stack/releases

Clone these repositories:  jgm/pandoc, jgm/texmath,
jgm/pandoc-types, jgm/pandoc-citeproc, in parallel
directories

Go into the pandoc directory and do

    git submodule update --init

Now make the needed modifications in pandoc-citeproc (to
Text/Pandoc/JSON), and then go back to the pandoc directory
and do

    stack install --install-ghc --stack-yaml=stack.full.yaml

**Comment by jgm** [comment]
Can someone who uses pandoc-citeproc on Windows confirm that the problem is fixed?
(Try the latest release, 0.9.1.1; this is included in the pandoc binary package installer.)

**Comment by jgm** [comment]
Closing.  If the problem can be reproduced with the latest version, please open a new issue.
